src/query_results.js

```diff
@@ -204,149 +204,148 @@ module.exports = class QueryResult {
+              initialQNodeIDToMatch = outputQNodeID;
 
               // like calling break in a loop
               return true;
             }
           }
         });
 
-        if (initialQueryEdgeID) {
+        if (initialQEdgeID) {
           // like calling break in a loop
           return true;
         }
       }
     });
 
-    debug(`initialQueryEdgeID: ${initialQueryEdgeID}, initialQueryNodeIDToMatch: ${initialQueryNodeIDToMatch}`);
+    debug(`initialQEdgeID: ${initialQEdgeID}, initialQNodeIDToMatch: ${initialQNodeIDToMatch}`);
 
-    // 'preresult' just means it has the data needed to assemble a result,
+    // 'queryGraphSolution' just means it has the data needed to assemble a result,
     // but it's formatted differently for easier pre-processing.
-    const preresults = [];
-    this._getPreresults(
-      dataByEdge,
-      initialQueryEdgeID,
-      edgeCount,
-      preresults,
-      [], // first preresult
-      initialQueryNodeIDToMatch,
+    const queryGraphSolutions = [];
+    this._getqueryGraphSolutions(
+      recordsByQEdgeID,
+      initialQEdgeID,
+      qEdgeCount,
+      queryGraphSolutions,
+      [], // first queryGraphSolution
+      initialQNodeIDToMatch,
     );
 
     /**
      * Consolidation
      *
      * With reference to this graphic:
      * https://github.com/biothings/BioThings_Explorer_TRAPI/issues/341#issuecomment-972140186
-     * The preresults are analogous to the collection of sets in the lower left. Now we want
-     * to consolidate the preresults as indicated by the the large blue arrow in the graphic
-     * to get consolidatedPreresults, which are almost identical the the final results, except
+     * The queryGraphSolutions are analogous to the collection of sets in the lower left. Now we want
+     * to consolidate the queryGraphSolutions as indicated by the the large blue arrow in the graphic
+     * to get consolidatedResults, which are almost identical the the final results, except
      * for some minor differences that make it easier to perform the consolidation.
      *
-     * There are two cases where we need to consolidate preresults:
+     * There are two cases where we need to consolidate queryGraphSolutions:
      * 1. one or more query nodes have an 'is_set' param
-     * 2. one or more primaryID pairs have multiple kgEdges each
+     * 2. one or more primaryCurie pairs have multiple kgEdges each
      *
-     * We perform consolidation by first grouping preresults by uniqueResultID and
-     * then merging each of those groups into a single consolidatedPreresult.
+     * We perform consolidation by first grouping queryGraphSolutions by uniqueResultID and
+     * then merging each of those groups into a single consolidatedResult.
      */
 
-    const preresultsByUniqueResultID = {};
-    preresults.forEach((preresult) => {
-      // example inputPrimaryID and outputPrimaryID in a preresult:
+    const solutionsByResultID = {};
```

Comment:
If we're using `uniqueResultID` elsewhere, should this be `solutionsByUniqueResultID`?

A `uniqueResultID` can be thought of as uniquely identifying a `consolidatedSolution`. Since each `consolidatedSolution` is associated in a 1:1 relationship with its corresponding TRAPI result, a `uniqueResultID` can also be thought of as uniquely identifying a TRAPI result, which is the actual goal. So how about changing to these names:
- `trapiResultID`
- `solutionsByTrapiResultID`
or if we understand `result` to just refer to TRAPI results, we could simply use:
- `resultID`
- `solutionsByResultID`

Reply:
I've switched to using `trapiResultID` as I think this is useful to make more obvious to unfamiliar devs.